From 5551e5315f77ad10d40b2b99c31f2281d81458d9 Mon Sep 17 00:00:00 2001
From: Ramasamy Kaliappan <quic_rkaliapp@quicinc.com>
Date: Mon, 1 Apr 2024 23:50:11 +0530
Subject: [PATCH] hostapd: ru puncturing is not working in channel switch

ru_punct_bitmap is used to set the puncturing bitmap in qsdk,
punct_bitmap is used in upstream. During package upgrades added
punct_bitmap which is not used by qsdk.

During channel swithing, punct_bitmap is used instead of ru_punct_bitmap
to check if the puncturing bitmap pattern is valid.

Add change to use ru_punct_bitmap instead of punct_bitmap in chan_switch.

Signed-off-by: Ramasamy Kaliappan <quic_rkaliapp@quicinc.com>
---
 hostapd/ctrl_iface.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hostapd/ctrl_iface.c b/hostapd/ctrl_iface.c
index 1fecd20..ae7b511 100644
--- a/hostapd/ctrl_iface.c
+++ b/hostapd/ctrl_iface.c
@@ -2942,7 +2942,7 @@ static int hostapd_ctrl_iface_chan_switch(struct hostapd_iface *iface,
         }
 
 	ret = hostapd_ctrl_check_freq_params(&settings.freq_params,
-					     settings.punct_bitmap);
+					     settings.freq_params.ru_punct_bitmap);
 	if (ret) {
 		wpa_printf(MSG_INFO,
 			   "chanswitch: invalid frequency settings provided");
-- 
2.34.1

